### ISSUE TITLE
invalid jfreechart dependency

### ISSUE THREAD

**Issue by Andrei-Pozolotin** [dwdyer/uncommons-maths#14]
@dwdyer Dan:

here 
https://repo1.maven.org/maven2/io/gatling/uncommons/maths/uncommons-maths/1.2.3/uncommons-maths-1.2.3.pom

you still depend on 

```
<groupId>jfree</groupId>
<artifactId>jfreechart</artifactId>
<version>1.0.8</version>
```

which is missing form maven central
http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22jfree%22%20AND%20a%3A%22jfreechart%22

**Comment by dwdyer** [comment]
The core library doesn't depend on JFreeChart, it's only used by the demo.  I don't know who put Uncommons Maths on Maven Central (pretty sure it wasn't me), but it's probably fine just to remove the dependency.

**Comment by slandelle** [comment]
@dwdyer I did, in my own project's groupId, as you didn't answer on #10 and I really needed 1.2.3 to be available on central.

**Comment by slandelle** [comment]
@dwdyer I already offered to help deploying on central in your own groupId, and my offer is still valid :)

**Comment by dwdyer** [comment]
@slandelle That's fine. Do you need any input from me to put it on there using a different group ID (and does it matter which group ID it's under if it's working already)? Maven support isn't a priority for me but I'm happy for you to do whatever you need to make things work for you.

Are you able to just drop the JFreeChart dependency for now or does that cause other problems?

**Comment by slandelle** [comment]
> does it matter which group ID it's under if it's working already

Well, uncommons-maths has already been deployed in `org.uncommons` AND `org.uncommons.maths` groupIds. Are you the one who deployed them there? In theory, as the project's owner, you should be the only one allowed to deploy there, or ask Sonatype to grant permissions for other people to do so.

I can of course keep on deploying in my own groupId. Less clean, would more advertise my project than yours (would make it appear as a subproject), but sure easier. Your call.

> Are you able to just drop the JFreeChart dependency for now or does that cause other problems?

@Andrei-Pozolotin The dependency is optional, so I wouldn't expect it to be an issue, even if it's missing. My project depends on it without any issue. What's your problem exactly?

**Comment by Andrei-Pozolotin** [comment]
people who did releases: 

Benson Margulies
https://issues.sonatype.org/browse/OSSRH-1565

Dan Dyer
https://issues.sonatype.org/browse/OSSRH-6063

@dwdyer it says it was you :-)

**Comment by Andrei-Pozolotin** [comment]
@slandelle it requires this much hacking to make it work, 
or else mvn fails with "artifact resolution" exception:

https://github.com/carrot-garden/scala_akka/blob/a34d0f6dc843acf0a40d8dab7b4c1fb0d4f03bb5/akka-samples/akka-sample-osgi-dining-hakkers/uncommons/pom.xml#L25

**Comment by Andrei-Pozolotin** [comment]
@dwdyer it would probably help to move demo to separate repo/project so not to confuse people

**Comment by slandelle** [comment]
@Andrei-Pozolotin In the 1.2.2 pom deployed in `org.uncommons.maths` groupId, jfreechart was mandatory. In the 1.2.3 pom I deployed in `io.gatling.uncommons.maths`, it's optional as it should be. As a workaround, use this on and your build won't fail.

@dwdyer You could open an issue on Sonatype, just like in [OSSRH-6063](https://issues.sonatype.org/browse/OSSRH-6063) and ask them to grant me permissions to deploy in `org.uncommons.maths` groupId and I can take care of the rest. WDYT?

**Comment by Andrei-Pozolotin** [comment]
@slandelle yes, you are right, but we'd rather have you do a proper release in proper name space :-)
there is one more issue: we need proper OSGI headers. please do not forget them!

**Comment by Andrei-Pozolotin** [comment]
re: osgi
https://www.assembla.com/spaces/akka/tickets/2990#/activity/ticket:

**Comment by slandelle** [comment]
@Andrei-Pozolotin You misunderstood me: I can take care of deploying on central, if @dwdyer is willing to, but I have no interest in OSGI, so if you want the headers, please provide a Pull Request that @dwdyer will merge.

**Comment by Andrei-Pozolotin** [comment]
@slandelle no, you misunderstood me :-)
osgi is already in the build
https://github.com/dwdyer/uncommons-maths/blob/master/uncommons-maths.bnd
please take a look inside the final jar that manifest has the osgi headers. if not - ping me.
thank you.
